api/routes/onboarding.py

```diff
@@ -10,20 +16,85 @@ def onboarding():
     """
     docstring
     """
-    return "Onboarding API. This API endpoint is used for onboarding users and stokvels ( on creation )"
-
+    return render_template("onboarding_template.html")
 
 @onboarding_bp.route(f"{BASE_ROUTE}/users", methods=["POST"])
 def onboard_user():
     """
-    docstring
+    Handles onboarding of a new user.
     """
-    return "Complete Onboarding endpoint"
+    try:
+        name = request.form["name"]
+        surname = request.form["surname"]
+        cell_number = request.form["cellphone_number"]
+        id_number = request.form["id_number"]
+
+        flag = queries.check_if_number_exists_sqlite(cell_number)
+        print(f"Check if number exists: {flag}")
+
+        if not flag:
+            user = User(name=name, surname=surname, cell_number=cell_number, id_number=id_number, wallet_id="")
+            wallet = Wallet(user.id_number, user_wallet="ILP_test_string", user_balance=100)
+
+            try:
```

Comment:
No need to have nested Try statements, a single try will catch any exception that happens within its block, will just have to be specific with the excpetions you would like to catch

Reply:
For some reason I was being rerouted to an error page when I had one try/catch block, the nested try statements were the only ones that worked - but yes, I'll revisit and try again.